src/functional/ffront/ast_passes/single_static_assign.py

```diff
@@ -48,11 +58,17 @@ def foo():
         a__2 = 3 + a__1
         return a__2
 
-    Note that each variable name is assigned only once and never updated / overwritten.
+    Note that each variable name is assigned only once (per branch) and never updated / overwritten.
```

Comment:
I think it's needed, because it breaks the assumption, that there's only ever one assignment per variable.

```
if True:
   a__0 = 1
else:
   # Having a second assignment anywhere else to the same variable wasn't possible before
   a__0 = 2
```

I'm unsure if any of the following passes relied on this.

It's a bit unclear, because the original sentence never clarified whether a variable _is assigned only once_ during an execution or within the source code.